db/db_impl.cc

```diff
@@ -5106,6 +5108,14 @@ void DBImpl::MaybeFlushColumnFamilies() {
 
   unable_to_flush_oldest_log_ = false;
 
+  // Another thread could have removed some logs from alive_log_files_ while
+  // we were in this method. This is possible because SwitchMemtable() unlocks
+  // and re-locks mutex_.
+  if (alive_log_files_.begin()->number != oldest_alive_log) {
+    assert(alive_log_files_.begin()->number > oldest_alive_log);
+    return;
+  }
+
```

Comment:
@reidHoruff is it possible to move line 5119to 5131 before line 5082, just like before your previous change? Epecially for the non-2pc case, it feels safer to me.